daktari/checks/direnv.py

```diff
@@ -0,0 +1,43 @@
+import re
+
+from daktari.command_utils import get_stdout
+
+from daktari.os import OS
+from daktari.check import Check, CheckResult
+from daktari.version_utils import get_simple_cli_version
+from typing import Optional
+
+
+class DirenvInstalled(Check):
+    name = "direnv.installed"
+
+    def __init__(self, required_version: Optional[str] = None, recommended_version: Optional[str] = None):
+        self.required_version = required_version
+        self.recommended_version = recommended_version
+        self.suggestions = {
+            OS.OS_X: "<cmd>brew install direnv</cmd>",
+            OS.GENERIC: "Install direnv: https://direnv.net/#getting-started",
+        }
+
+    def check(self) -> CheckResult:
+        installed_version = get_simple_cli_version("direnv")
+        return self.validate_semver_expression(
+            "direnv", installed_version, self.required_version, self.recommended_version
+        )
+
+
+class DirenvAllowed(Check):
+    name = "direnv.allowed"
+    depends_on = [DirenvInstalled]
+
+    def __init__(self, parent_file_name: str):
+        self.parent_file_name = parent_file_name
```

Comment:
Do we need `parent_file_name`, do you think? Could we get away without it?

Reply:
Just tried it, looks like there are two cases:
1. Directory and parent both have .envrc files. `direnv status` finds the current directory's .envrc and reports its status, then stops. Parent directory not mentioned.
2. Directory does not have .envrc file, parent does. `direnv status` finds the parent directory's .envrc file and reports its status, then stops. 

So I guess to be completely thorough without passing through a `parent_file_name`, you could
1. Check direnv is installed
2. Check `./.envrc` exists
3. Check like in the current code, but without the `parent_file_name`

Reply:
Would it be simper perhaps not to worry about `parent_file_name`, and just have it run the check on the CWD?

Reply:
Ah that is much nicer!